dynparquet/hashed.go

```diff
@@ -105,19 +105,59 @@ func HashArray(arr arrow.Array) []uint64 {
 func hashListArray(arr *array.List) []uint64 {
 	res := make([]uint64, arr.Len())
 	digest := xxhash.New()
-
-	dictionaryList, binaryDictionaryList, err := arrowutils.ToConcreteList(arr)
-	if err != nil {
-		panic(err)
+	var hv func(int, *xxhash.Digest)
+	switch e := arr.ListValues().(type) {
+	case *array.Int64:
+		var buf [8]byte
+		hv = func(i int, d *xxhash.Digest) {
+			binary.BigEndian.PutUint64(buf[:], uint64(e.Value(i)))
+			_, _ = d.Write(buf[:])
+		}
+	case *array.Float64:
+		var buf [8]byte
+		hv = func(i int, d *xxhash.Digest) {
+			binary.BigEndian.PutUint64(buf[:], math.Float64bits(e.Value(i)))
+			_, _ = d.Write(buf[:])
+		}
+	case *array.Boolean:
+		var buf [1]byte
+		hv = func(i int, d *xxhash.Digest) {
+			if e.Value(i) {
+				buf[0] = 2
+			} else {
+				buf[0] = 1
+			}
+			_, _ = d.Write(buf[:])
+		}
+	case *array.Binary:
+		hv = func(i int, d *xxhash.Digest) {
+			_, _ = d.Write(e.Value(i))
+		}
+	case *array.String:
+		hv = func(i int, d *xxhash.Digest) {
+			_, _ = d.WriteString(e.Value(i))
+		}
+	case *array.Dictionary:
+		switch dict := e.Dictionary().(type) {
+		case *array.Binary:
+			hv = func(i int, d *xxhash.Digest) {
+				_, _ = d.Write(dict.Value(e.GetValueIndex(i)))
+			}
```

Comment:
@thorfour here is where we are covering the case before this change( list of binary dictionary).